### PR TITLE
Add cabal-install-3.4.0.0

### PR DESCRIPTION
**Pull request by RyanGlScott**

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Oops, I didn't realize that this grabs Windows `cabal-install` binaries from Chocolatey instead of from the `Cabal` website. In that case, this is blocked on Mistuke/CabalChoco#3.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -37,14 +37,14 @@ jobs:
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
         ghc: ["latest", "8.4.4"]
-        cabal: ["latest", "3.2.0.0", "3.4.0.0-rc4"]
+        cabal: ["latest", "3.2.0.0", "3.4.0.0"]
         expect-fail: [false]
         exclude:
           - os: windows-latest
-            cabal: "3.4.0.0-rc4"
+            cabal: "3.4.0.0"
```
Comment (jared-w) [inline .github/workflows/workflow.yml:44]: I actually have no idea what happens if you exclude windows with cabal 3.4 and then include it again.

That said, the include/exclude for windows here was intended to test cabal 3.4 so they can both probably be removed now that 3.4 is out "for real"

Comment (RyanGlScott) [reply .github/workflows/workflow.yml]: Good catch! I've pushed an update which removes these.

**Comment by RyanGlScott** [comment]
`cabal-install-3.4.0.0` is [now available on Chocolatey](https://chocolatey.org/packages/cabal/3.4.0.0). I restarted the CI and everything passes now!

**Pull request merged**